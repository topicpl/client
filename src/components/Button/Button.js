import React from 'react';

var buttonStyle = {
  margin: '10px 10px 10px 0'
};

const Button = (props) => {
    return (
      <button {...props} style={buttonStyle} onClick={props.handleClick}>{props.children}</button>
    );
}


export default Button;