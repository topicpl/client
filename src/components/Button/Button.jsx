import React from 'react';

const buttonStyle = {
  margin: '10px 10px 10px 0',
};

const Button = (props) => {
  const { handleClick, children } = props;
  return <button {...props} style={buttonStyle} onClick={handleClick}>{children}</button>;
};


export default Button;
