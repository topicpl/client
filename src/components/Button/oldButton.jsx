import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import themes from 'styled-theming';
import PropTypes from 'prop-types';

const backgroundBorderColor = themes('variant', {
  default: ({ theme, disabled }) =>
    disabled ? theme.color.grey[500] : theme.color.grey[400],
  success: ({ theme, disabled }) =>
    disabled ? theme.color.green[800] : theme.color.green[500],
});

const color = themes('variant', {
  default: ({ theme, disabled }) =>
    disabled ? theme.color.grey[800] : theme.color.black,
  success: ({ theme, disabled }) =>
    disabled ? theme.color.grey[500] : theme.color.white,
});

// hover
const hoverBorderColor = themes('variant', {
  default: ({ theme, disabled }) =>
    disabled ? theme.color.grey[500] : theme.color.black,
  success: ({ theme, disabled }) => theme.color.green[disabled ? 800 : 500],
});

const hoverBgColor = themes('variant', {
  default: ({ theme, disabled }) =>
    disabled ? theme.color.grey[500] : theme.color.grey[200],
  success: ({ theme, disabled }) => theme.color.green[disabled ? 800 : 600],
});

const hoverColor = themes('variant', {
  default: ({ theme, disabled }) =>
    disabled ? theme.color.grey[800] : theme.color.black,
  success: ({ theme, disabled }) => theme.color.grey[disabled ? 500 : 400],
});

const ButtonStyles = styled.button`
  cursor: pointer;
  font-size: 16px;
  border-radius: 3px;
  padding: 0.25em 1em;
  transition: 0.5s all ease-out;
  background: ${backgroundBorderColor};
  border: 1px solid ${backgroundBorderColor};
  color: ${color};
  font-weight: ${({ theme }) => theme.font.weight.medium};
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};

  &:hover {
    background: ${hoverBgColor};
    border-color: ${hoverBorderColor};
    color: ${hoverColor};
  }
`;

ButtonStyles.defaultProps = {
  variant: 'default',
};

ButtonStyles.propTypes = {
  variant: PropTypes.oneOf(['default', 'success']),
};

const oldButton = ({
  children,
  isLoading,
  disabled,
  variant = 'default',
  ...rest
}) => {
  const loadingComponent = 'Loading...';
  return (
    <ThemeProvider theme={{ variant }}>
      <ButtonStyles {...rest} disabled={disabled || isLoading}>
        {isLoading ? loadingComponent : children}
      </ButtonStyles>
    </ThemeProvider>
  );
};

export default oldButton;
