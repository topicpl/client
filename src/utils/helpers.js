export const getQueryVariable = (variable) => {
  const query = window.location.search.substring(1);
  const vars = query.split('&');
  const varsWithValues = vars.reduce((acc, pair) => {
    const [key, value] = pair.split('=');
    return { ...acc, [decodeURIComponent(key)]: decodeURIComponent(value) };
  }, {});

  return varsWithValues[variable] || null;
};
