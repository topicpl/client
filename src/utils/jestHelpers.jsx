
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { theme } from '../assets/styles/theme';

export const mountWithTheme = (children) => renderer.create(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
