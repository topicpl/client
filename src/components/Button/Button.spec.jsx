import React from 'react';
import Button from './Button';
import { mountWithTheme } from '../../utils/jestHelpers';

describe('Button', () => {
  test('should match variant - default', () => {
    const wrapper = mountWithTheme(<Button>my button</Button>);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should match variant - success', () => {
    const wrapper = mountWithTheme(<Button variant="success">my button</Button>);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
