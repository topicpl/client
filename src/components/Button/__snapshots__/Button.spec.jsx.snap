// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match variant - default 1`] = `
<button
  className="Button__ButtonStyles-bFYiJu bBNfOj"
>
  my button
</button>
`;

exports[`Button should match variant - success 1`] = `
<button
  className="Button__ButtonStyles-bFYiJu ewQbZy"
>
  my button
</button>
`;
